lib/puppet/type/service.rb

```diff
@@ -73,6 +73,17 @@ def retrieve
         provider.enabled?
       end
 
+      # This only makes sense on systemd systems. Static services cannot be enabled
+      # or disabled manually.
+      def insync?(current)
+        if provider.respond_to?(:cached_enabled?) && provider.cached_enabled? == 'static'
```

Comment:
This could be abstracted to something, like `unenforcable?`. However, I don't see a strong need for it yet.